tests/random_test.py

```diff
@@ -639,14 +639,19 @@ def testShuffle(self, dtype):
   def testChoice(self, dtype, input_range_or_shape, shape, replace, weighted, axis):
     # This is the function API that we test against (note that self.rng().choice differs)
     np_choice = np.random.default_rng(0).choice
+    p_dtype = dtypes._to_inexact_dtype(dtype)
```

Comment:
Out of curiosity, why is this needed?

Reply:
To make the test cover the failure in the attached issue, we needed to use a non-default dtype for `p`. This seemed like a clean way to do it while maintaining coverage